### PR TITLE
`Bugfix`: Removed moderator rights to edit posts; added test

### PR DESCRIPTION
**Pull request by FelberMartin**
### Problem Description
Currently a user with moderation rights can edit anyone's posts. As discussed in the Communication Team Meeting, this behaviour is unwanted.

### Changes
 
- Removed moderator rights to edit other's posts
- Created unit test

### Steps for testing
- Login as an instructor (has moderation rights)
- Long press another users post
- See that "Edit" is not available anymore
- Press on a post of yourself 
- "Edit" is still possible

### CODE REVIEW

**Review by julian-wls — APPROVED** [review]
Code lgtm. Works as expected!
As an instructor, I can only edit my own posts, but not others posts.

File: feature/metis/conversation/src/main/kotlin/de/tum/informatics/www1/artemis/native_app/feature/metis/conversation/ui/post/PostActions.kt [hunk 1]
```diff
@@ -46,24 +45,24 @@ fun rememberPostActions(
...
+            return@remember PostActions()
         }
+
+        val doesPostExistOnServer = post.serverPostId != null
+        val hasEditPostRights = post.authorId == clientId
+        val hasResolvePostRights = isAtLeastTutorInCourse || post.authorId == clientId
+
+        PostActions(
+            requestEditPost = if (doesPostExistOnServer && hasEditPostRights) onRequestEdit else null,
+            requestDeletePost = if (hasEditPostRights) onRequestDelete else null,
```
Comment (julian-wls) [inline feature/metis/conversation/src/main/kotlin/de/tum/informatics/www1/artemis/native_app/feature/metis/conversation/ui/post/PostActions.kt:58]: `hasEditPostRights` is used for deleting, too. However, deleting should be possible for moderators. Could you adjust this?

**Review by julian-wls — CHANGES_REQUESTED** [review]
Sorry I should have looked better, I missed one crucial thing, see below.

File: feature/metis/conversation/src/main/kotlin/de/tum/informatics/www1/artemis/native_app/feature/metis/conversation/ui/post/PostActions.kt (hunk 1 above)
Comment (FelberMartin) [reply feature/metis/conversation/src/main/kotlin/de/tum/informatics/www1/artemis/native_app/feature/metis/conversation/ui/post/PostActions.kt:58]: Very true, good catch!

**Review by julian-wls — APPROVED** [review]

**Review by TimOrtel — CHANGES_REQUESTED** [review]
The code looks good but I think it would be better if you added UI tests. We should have tests that actually view some posts, long click a post and assert that the edit button is available or not in the bottom sheet.

**Comment by julian-wls** [comment]
I think the UI test would definitely make sense; I'm not sure, though, if we should include them as part of this PR. The changes were discussed in the last meeting and seem to be urgent. I would include them in the upcoming release and create an issue for the UI tests. Furthermore, there seems to be an ongoing issue with the RobolectricRunner and the modal bottom sheet, which is another reason to postpone the UI tests.

**Pull request merged**